### ISSUE TITLE
SystemZ Z13 Unsupported vector argument or return type

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#41420]
|  |  |
| --- | --- |
| Bugzilla Link | [42075](https://llvm.org/bz42075) |
| Version | trunk |
| OS | FreeBSD |
| Attachments | [A file that, when run, should exemplify the bug.](https://user-images.githubusercontent.com/60944935/143759300-f11d71c3-6f4b-435a-a2fa-90b407fbc50d.gz) |
| Reporter | LLVM Bugzilla Contributor |
| CC | @uweigand |

## Extended Description 

For some reason, when llc is run with options -mtriple=s390x -mcpu=z13 (without the second argument, the issue does not occur), llc prints "LLVM ERROR: Unsupported vector argument or return type" if the input file contains any function whose arguments or return types are, particularly v3f64, but also generally any vector type that is considered to be an extended type, such as v6f32.  

It fails at an assert at VerifyVectorType in lib/Target/SystemZ/SystemZISelLowering.cpp.  As far as I can tell, the code seems to run to completion and produce correct assembly as long as this assert is simply skipped in those cases.  However, there is a comment indicating that a single element vector containing a 128 bit type is not supporting, suggesting that the assert was meant to catch this case but also ends up catching many other cases that it is not supposed to.

I have attached a test file.

**Comment by dominik-steenken** [comment]
I do not think this is an issue anymore. The provided repro runs fine, and the types mentioned are handled properly. `VerifyVectorType` does not even seem to exist anymore.

**Comment by uweigand** [comment]
Yes, this has been fixed in the meantime.
